src/arrayMethodJoin.js

```diff
@@ -4,8 +4,33 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let result = '';
+
+    for (let i = 0; i < this.length; i++) {
+      switch (true) {
+        case i === this.length - 1 && this[i] === undefined:
+          break;
+
+        case i === this.length - 1:
+          result += this[i];
+          break;
+
+        case separator === null:
+          result += this[i] + 'null';
```

Comment:
Why should you add word 'null' to your string?
Btw, you have default value for your separator, so it can't be null

Reply:
thank you!